mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -566,4 +627,28 @@ mod tests {
 
         assert_eq!(expected_stake_distribution, computed_stake_distribution);
     }
+
+    #[tokio::test]
+    async fn get_current_kes_period() {
+        let socket_path = create_temp_dir("get_current_kes_period").join("node.socket");
```

Comment:
You can use the new `mithril_common::test_utils::TempDir` module for creating safe socket paths.

Reply:
It already uses it

```rust
/// Creates a new work directory in the system's temporary folder.
fn create_temp_dir(folder_name: &str) -> PathBuf {
    TempDir::create_with_short_path("pallas_chain_observer_test", folder_name)
}
```